### PR TITLE
[py-gevent] Fix: deprecated compiler flags

### PR DESCRIPTION
**Pull request by stephenmsachs**
Upstream PR: https://github.com/gevent/gevent/pull/1896

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-gevent/package.py [hunk 1]
```diff
@@ -35,3 +35,7 @@ class PyGevent(PythonPackage):
     depends_on('py-greenlet@0.4.13:', when='@:1.4', type=('build', 'run'))
     depends_on('py-zope-event', when='@20.5.1:', type=('build', 'run'))
     depends_on('py-zope-interface', when='@20.5.1:', type=('build', 'run'))
+
+    patch('icc.patch',
+          sha256='44a3aa7ec3a9f056145c25492012f494166dee4c816a3a998a7e3b2d85b57c77',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gevent/package.py]: No need for a sha256 if it's a local file. Also, does this patch apply to all versions of the package?

Comment (stephenmsachs) [reply var/spack/repos/builtin/packages/py-gevent/package.py]: I double checked, it works with all versions.

I had to add the sha to make it work at first. It works without sha now on a fresh Spack installation.

File: var/spack/repos/builtin/packages/py-gevent/package.py [hunk 2]
```diff
@@ -35,3 +35,5 @@ class PyGevent(PythonPackage):
     depends_on('py-greenlet@0.4.13:', when='@:1.4', type=('build', 'run'))
     depends_on('py-zope-event', when='@20.5.1:', type=('build', 'run'))
     depends_on('py-zope-interface', when='@20.5.1:', type=('build', 'run'))
+
+    patch('icc.patch', when='%intel')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-gevent/package.py:40]: Can we add a brief description of what the patch does and where does it come from, just above the directive? Other than that LGTM.

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-gevent/package.py (hunk 2 above)
Comment (stephenmsachs) [reply var/spack/repos/builtin/packages/py-gevent/package.py:40]: Done

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**